### PR TITLE
Add MuP optimizers

### PR DESCRIPTION
**Pull request by mrwyattii**
Integrating the optimizers from the [MuP](https://github.com/microsoft/mup) project.

### CODE REVIEW

File: tests/unit/test_mup_optimizers.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation.
```
Comment (loadams) [inline tests/unit/test_mup_optimizers.py:1]: What folder should this go into inside of `tests/unit`?

Comment (mrwyattii) [reply tests/unit/test_mup_optimizers.py:1]: Probably `unit/runtime/`, but maybe in one of the subdirs. Look for where we have other tests related to optimizers!

Comment (loadams) [reply tests/unit/test_mup_optimizers.py]: Moved to runtime.  The other optimizers are ops, so they are in folders there.  So just moving this to runtime for now.

**Review by mrwyattii — COMMENTED** [review]
LGTM, thanks @loadams

**Review by loadams — APPROVED** [review]

**Pull request merged**